src/main/java/org/terasology/logic/health/DamageAuthoritySystem.java

```diff
@@ -103,32 +106,55 @@ private void doDamage(EntityRef entity, int damageAmount, Prefab damageType, Ent
         if ((health != null) && !ghost) {
             int cappedDamage = Math.min(health.currentHealth, damageAmount);
             health.currentHealth -= cappedDamage;
-            entity.send(new ActivateRegenEvent());
             entity.saveComponent(health);
             entity.send(new OnDamagedEvent(damageAmount, cappedDamage, damageType, instigator));
             if (health.currentHealth == 0 && health.destroyEntityOnNoHealth) {
                 entity.send(new DestroyEvent(instigator, directCause, damageType));
             }
+            scheduleRegenEvent(entity, health.waitBeforeRegen);
+        }
+    }
+
+    private void scheduleRegenEvent(EntityRef entity, float delayInSeconds) {
+        // deactivate base regen because entity was damaged
+        entity.send(new DeactivateRegenEvent());
+        // reset timer for activating the base regen on the entity
+        if (delayManager.hasDelayedAction(entity, DELAYED_REGEN_ACTIVATION)) {
+            logger.debug("Canceling previous delayed regen event");
+            delayManager.cancelDelayedAction(entity, DELAYED_REGEN_ACTIVATION);
+        }
+        long delayInMs = Math.round(delayInSeconds * 1000);
+        logger.debug("Scheduling delayed regen event with delay '{}'", delayInMs);
+        delayManager.addDelayedAction(entity, DELAYED_REGEN_ACTIVATION, delayInMs);
+    }
+
+    @ReceiveEvent
+    public void onDelayedRegenActivation(DelayedActionTriggeredEvent event, EntityRef entity, HealthComponent health) {
+        if (event.getActionId().equals(DELAYED_REGEN_ACTIVATION)) {
+            logger.debug("Sending delayed regen event");
```

Comment:
```suggestion
            logger.debug("Sending regen event");
```
or
```suggestion
            logger.debug("Sending regen event after delay has passed");
```

because it's no longer delayed now.